assignments/homework/pcosta/497HW2.py

```diff
@@ -0,0 +1,124 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from astropy.io import fits
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy.table import QTable
+from astropy.table import Table
+
+#Notebook 4
+#1
+gaiaLMC = fits.open('../../../astropy_notebooks/data/gaia_lmc_psc.fits')
+firstHDUHeader = gaiaLMC[0].header
+secondHDUHeader = gaiaLMC[1].header
+print(firstHDUHeader)
+print(secondHDUHeader)
+firstHDUHeader['BIGFISH'] = 'Always.'
+secondHDUHeader['TWO'] = "No more, no less."
+
+#2
+data = gaiaLMC[1].data
+gBandMag = data['phot_g_mean_mag']
+num_bins = 25
+n, bins, patches = plt.hist(gBandMag, num_bins, facecolor='blue')
+plt.show()
+maxGmag = np.amax(data['phot_g_mean_mag'])
+print("Upper limit for Gmag:",maxGmag)
+
+#3
+plotCoords = SkyCoord(data['ra']*u.deg, data['dec']*u.deg, frame='fk5')
+plotGalactic = plotCoords.transform_to('galactic')
+lCoord = plotGalactic.l
+bCoord = plotGalactic.b
+plt.plot(lCoord, bCoord, 'o', markersize=2)
+plt.xlabel("l Coordinate (Degrees)")
+plt.ylabel("b coordinate (Degrees)")
+plt.show()
+
+#4
+hduImage = fits.ImageHDU(fits.open('../../../astropy_notebooks/data/LMCDensFits1k.fits'))
+hduTable = fits.BinTableHDU(fits.open('../../../astropy_notebooks/data/gaia_lmc_psc.fits'))
+hduImage.writeto('myFits.fits')
+hduTable.writeto('myFits.fits')
+
+
+#Notebook 5
+#1.1
+table1 = QTable()
+table1['Spectral Type'] = ['O', 'B', 'A', 'F', 'G']
+table1['Temperature'] = [40000, 20000,8500, 6500, 5700]
+table1['Temperature'].unit = u.K
+table1['Radius'] = [10, 5, 1.7, 1.3, 1.0]
+table1['Radius'].unit = u.Msun
+print(table1)
+
+#1.2
+meanTemp = np.mean((np.array(table1['Temperature'])))*u.K
```

Comment:
No need to all these conversions. The point of `QTable`/Quantities is that the numpy functions work natively:

np.mean(table1['Temperature'])